docs/src/submodules/Utilities/overview.md

```diff
@@ -491,3 +491,18 @@ function MOI.get(model::Optimizer, attr::MOI.ObjectiveFunction)
     return MOI.Utilities.get_fallback(model, attr)
 end
 ```
+
+## DoubleDicts
+
+When writing MOI interfaces, we often need to handle situations in which we map
+[`ConstraintIndex`](@ref)s to different values. For example, to a string
+for [`ConstraintName`](@ref).
+
+One option is to use a dictionary like `Dict{MOI.ConstraintIndex,String}`.
+However, this incurs a performance cost because the key is not a concrete type.
```

Comment:
It's not clear what's the gain is here. The gain occurs only when you need to use the dictionary many times for the same set-function pair. Then DoubleDict is useful as `dict[F, S]` will give a dictionary with concrete key type. Bu this should be coupled with a function barrier.